File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        #if(s[0]==' ' and len(s)-1==' '):
            
        #s.strip(' ')
        print(s)
        li=s.split()
        n=len(li)
        i=0
        temp=""
        print(li)
        start=0
        end=n-1
        while(start<=end):
            temp=li[start]
            li[start]=li[end]
            li[end]=temp
            start+=1
            end-=1
        print(li)
        x = " ".join(li)
        return x.strip()
        
        
        

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int left = 0; 
        int n = s.length();
        string ans;
        string temp;
        
        while(left<n) {
            char ch = s[left];
            if (ch == ' ') {
                if (temp != "") {
                    if (ans == "") {
                        ans = temp;
                    } else {
                        ans = temp + " " + ans;
                    }
                    temp.clear();
                } 
            } else {
                temp+=ch;
            }
            left++;
        }
        
        if (temp!="") { // for last word
            if (ans!="") {
                ans = temp + " " + ans;
            } else {
                ans = temp;
            }
        }
        return ans;
    }
};